provider/infoblox/infoblox_test.go

```diff
@@ -343,6 +343,52 @@ func createMockInfobloxObject(name, recordType, value string) ibclient.IBObject
 	return nil
 }
 
+func createMockInfobloxObject2(name, recordType, value string, value2 string) ibclient.IBObject {
+	ref := fmt.Sprintf("record:%s/%s:%s/default", strings.ToLower(recordType), base64.StdEncoding.EncodeToString([]byte(name)), name)
+	switch recordType {
+	case endpoint.RecordTypeA:
+		return ibclient.NewRecordA(
+			ibclient.RecordA{
+				Ref:      ref,
+				Name:     name,
+				Ipv4Addr: value,
+			},
+		)
+	case endpoint.RecordTypeCNAME:
+		return ibclient.NewRecordCNAME(
+			ibclient.RecordCNAME{
+				Ref:       ref,
+				Name:      name,
+				Canonical: value,
+			},
+		)
+	case endpoint.RecordTypeTXT:
+		return ibclient.NewRecordTXT(
+			ibclient.RecordTXT{
+				Ref:  ref,
+				Name: name,
+				Text: value,
+			},
+		)
+	case "HOST":
+		return ibclient.NewHostRecord(
+			ibclient.HostRecord{
+				Ref:  ref,
+				Name: name,
+				Ipv4Addrs: []ibclient.HostRecordIpv4Addr{
+					{
+						Ipv4Addr: value,
+					},
+					{
+						Ipv4Addr: value2,
+					},
+				},
+			},
+		)
+	}
+	return nil
+}
+
```

Comment:
```suggestion
```
my suggestion is to remove this function and update `createMockInfobloxObject` with a variadict.
* less code
* more control
```go
func createMockInfobloxObject(name string, recordType string, values ...string) ibclient.IBObject {
	ref := fmt.Sprintf("record:%s/%s:%s/default", strings.ToLower(recordType), base64.StdEncoding.EncodeToString([]byte(name)), name)
	switch recordType {
	case endpoint.RecordTypeA:
		return ibclient.NewRecordA(
			ibclient.RecordA{
				Ref:      ref,
				Name:     name,
				Ipv4Addr: values[0],
			},
		)
	case endpoint.RecordTypeCNAME:
		return ibclient.NewRecordCNAME(
			ibclient.RecordCNAME{
				Ref:       ref,
				Name:      name,
				Canonical: values[0],
			},
		)
	case endpoint.RecordTypeTXT:
		return ibclient.NewRecordTXT(
			ibclient.RecordTXT{
				Ref:  ref,
				Name: name,
				Text: values[0],
			},
		)
	case "HOST":
		var ipv4Addrs []ibclient.HostRecordIpv4Addr
		for _, value := range values {
			ipv4Addrs = append(ipv4Addrs, ibclient.HostRecordIpv4Addr{
				Ipv4Addr: value,
			})
		}

		return ibclient.NewHostRecord(
			ibclient.HostRecord{
				Ref:       ref,
				Name:      name,
				Ipv4Addrs: ipv4Addrs,
			},
		)
	}
	return nil
}
```

Reply:
Commit-ed suggestion.